shared/api/c_api_tokenizer.cc

```diff
@@ -47,6 +47,51 @@ extError_t ORTX_API_CALL OrtxCreateTokenizer(OrtxTokenizer** tokenizer, const ch
   return status.Code();
 }
 
+extError_t ORTX_API_CALL OrtxCreateTokenizerWithOptions(
+    OrtxTokenizer** tokenizer,
+    const char* tokenizer_path,
+    const char* keys[],
+    const char* values[],
+    size_t num_options) 
+{
+  // Validate tokenizer_path
+  if (tokenizer_path == nullptr) {
+    ReturnableStatus::last_error_message_ = "The tokenizer data directory is null";
+    return kOrtxErrorInvalidArgument;
+  }
+
+  if (!path(tokenizer_path).is_directory()) {
+    ReturnableStatus::last_error_message_ =
+        std::string("Cannot find the directory of ") + tokenizer_path;
+    return kOrtxErrorInvalidArgument;
+  }
+
+  // Convert keys/values for tokenizer options to unordered_map
+  std::unordered_map<std::string, std::string> options;
+  for (size_t i = 0; i < num_options; ++i) {
+    if (keys[i] && values[i]) {
+      options[keys[i]] = values[i];
+    }
+  }
```

Comment:
Shall we check if an option was passed in that is not a valid option (in case of typos from caller side)?

Reply:
done!